public/app/plugins/panel/geomap/components/DataHoverView.tsx

```diff
@@ -24,31 +24,43 @@ export interface Props {
   header?: string;
 }
 
+interface DisplayValue {
+  name: string;
+  value: unknown;
+  valueString: string;
+  highlight: boolean;
+}
+
 export const DataHoverView = ({ data, rowIndex, columnIndex, sortOrder, mode, header = undefined }: Props) => {
   const styles = useStyles2(getStyles);
 
   if (!data || rowIndex == null) {
     return null;
   }
-
+  const fields = data.fields.map((f, idx) => {
+    return { ...f, hovered: idx === columnIndex };
+  });
   // Put the traceID field in front.
-  const visibleFields = data.fields.filter((f) => !Boolean(f.config.custom?.hideFrom?.tooltip));
-  const traceIDField = visibleFields.find((field) => field.name === 'traceID') || data.fields[0];
+  const visibleFields = fields.filter((f) => !Boolean(f.config.custom?.hideFrom?.tooltip));
+  const traceIDField = visibleFields.find((field) => field.name === 'traceID') || fields[0];
   const orderedVisibleFields = [traceIDField, ...visibleFields.filter((field) => traceIDField !== field)];
 
   if (orderedVisibleFields.length === 0) {
     return null;
   }
 
-  const displayValues: Array<[string, unknown, string]> = [];
+  const displayValues: DisplayValue[] = [];
   const links: Array<LinkModel<Field>> = [];
   const linkLookup = new Set<string>();
 
-  for (const f of orderedVisibleFields) {
-    const v = f.values.get(rowIndex);
-    const disp = f.display ? f.display(v) : { text: `${v}`, numeric: +v };
-    if (f.getLinks) {
-      f.getLinks({ calculatedValue: disp, valueRowIndex: rowIndex }).forEach((link) => {
+  for (const field of orderedVisibleFields) {
+    if (mode === TooltipDisplayMode.Single && columnIndex != null && !field.hovered) {
+      continue;
+    }
+    const value = field.values.get(rowIndex);
```

Comment:
Please note that I had to change this from `const value = field.values[rowIndex];` because values is not of type `any[] | Vector<any>` in v9.5.x